### ISSUE TITLE
Use of variables in type expression are sometimes not reported

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#8316]
This comes from a mypy issue: https://github.com/python/mypy/issues/17493

The following code should generate an error:

```python
from enum import Enum
from typing import Final, Literal

class Marker(Enum):
    unspecified = 0

unspecified: Final = Marker.unspecified

def func(value: int | Literal[unspecified] = unspecified) -> None:  # error: Parameter 1 of Literal[...] is invalid  [valid-type]
    ...
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.371.
